### PR TITLE
feat: create `Timestamp` filter

### LINKED ISSUE #434: Add filters to blobs, blocks and transactions pages
Create a filters bar on top of the tables in the `/blobs`, `/txs` and `/blocks` page views.

The bar should be comprised of the following filters: 

- [x] #465
- [ ] #466
- [ ] #467
- [ ] #468 (Show only in the `/blocks` page view

It should also have a primary button with the `filter` label. The button should be disable by default and only enable when one of the filters have been set.

The bar sholuld be completely responsive as well as each filter component

Mockup:

image

### LINKED ISSUE #466: Create a timestamp filter
Use the following [datepicker](https://react-tailwindcss-datepicker.vercel.app)

The filter should be comprised of two joined text fields separated by a vertical bar. When click either of them the date picker should be displayed. 

They should display the following hints: "Start date" and "End date"

### PR DESCRIPTION
**Pull request by xFJA**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [X] I have filled out the description and linked the related issues.

### Description

- Add to `Dropdown` a custom `height` prop to keep all inputs inside `FilterPanel` with same height
- Add `fullWidth` prop to `Button` to be used in small screens layouts
- Install library to have a Tailwind React `DatePicker` component
- Add a `timestamp` dropdown to be able to filter in the `blobs`, `blocks`, and `transactions` view tables by start and end date

### Related Issue 

- Closes #466  
- Closes #434 

### Screenshots 

image

### CODE REVIEW

**Review by PabloCastellano — COMMENTED** [review]
I tried this PR in the Vercel preview, and here are some notes:

- The filter doesn't actually seem to work (and the Filter button is always gray (is it disabled?)).
- Is the number of blob transactions updated after the filters are applied? From your screenshot, it looks like it didn't get updated accordingly.

**Comment by xFJA** [comment]
> I tried this PR in the Vercel preview, and here are some notes:
> 
> * The filter doesn't actually seem to work (I select dates and nothing happens, also the Filter button is always gray (is it disabled?)).
> * Is the number of blob transactions updated after the filters are applied? From your screenshot, it looks like it didn't get updated accordingly.

The branch was missing a `git rebase` because many files in the target branch have been refactored! You can check now :+1:

**Review by PabloCastellano — COMMENTED** [review]
I've tried again and picking the dates won't let me click the Filter button.

We should also allow users to filter by date range, without selecting any particular rollup.

**Comment by xFJA** [comment]
> I've tried again and picking the dates won't let me click the Filter button.
> 
> We should also allow users to filter by date range, without selecting any particular rollup.

The conditions to enable the filter button has been reduced to at least one filled. Now you can filter using the timestamp without selecting the rollup

**Review by PabloCastellano — COMMENTED** [review]
Some bugs still present:

- The number of blob transactions is not updated after the filters are applied
- It is impossible to remove the date range once selected due to the button being disabled when I remove the dates.

**Comment by xFJA** [comment]
> Some bugs still present:
> 
> * The number of blob transactions is not updated after the filters are applied
> * It is impossible to remove the date range once selected due to the button being disabled when I remove the dates.

Added a new statement to check whether all the filters have been reset individually. If `true`, the table is reset to all items automatically :+1:

**Review by PJColombo — APPROVED** [review]
Merged rollup filter latest changes and aligned the styles of the dropdown and date picker

**Review by luis-herasme — APPROVED** [review]

File: apps/web/src/hooks/useQueryParams.ts [hunk 1]
```diff
@@ -6,10 +6,12 @@ import { Rollup } from "@blobscan/api/enums";
 import type { Rollup as LowercaseRollup } from "~/types";
 
 type QueryParams = {
+  from?: string;
   p: number;
   ps: number;
   rollup?: LowercaseRollup | "null";
```
Comment (xFJA) [inline apps/web/src/hooks/useQueryParams.ts:12]: I think the `rollup` shouldn´t be a query param as we use the `from` mapped address associated instead

File: apps/web/src/utils/date.ts [hunk 2]
```diff
@@ -6,6 +6,14 @@ export function normalizeTimestamp(timestamp: number | Date | string) {
     : dayjs(timestamp);
 }
 
+export function getISODate(date: dayjs.Dayjs | Date | string) {
```
Comment (xFJA) [inline apps/web/src/utils/date.ts:9]: should we add UTs here?

File: packages/api/src/app-router.ts [hunk 3]
```diff
@@ -32,6 +33,7 @@ export const appRouter = t.router({
     .input(z.void())
     .output(z.string())
     .query(() => "yay!"),
+  getRollups,
```
Comment (xFJA) [inline packages/api/src/app-router.ts:36]: These changes were removed in the rollup filter refactor. This should be an error in managing conflicts after many merges

File: packages/api/src/getRollups.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Rollup } from "@blobscan/db";
```
Comment (xFJA) [inline packages/api/src/getRollups.ts:1]: These changes were removed in the rollup filter refactor. This should be an error in managing conflicts after many merges

**Review by xFJA — COMMENTED** [review]
It´s seems the background color overriding for the date range picker component has dissappeared in the refactor because the colors now mismatch: 

image

File: apps/web/src/hooks/useQueryParams.ts (hunk 1 above)
Comment (PJColombo) [reply apps/web/src/hooks/useQueryParams.ts:12]: Actually, we still need it in order to filter non-rollup elements (`rollup=null`)

**Comment by PJColombo** [comment]
> It´s seems the background color overriding for the date range picker component has dissappeared in the refactor because the colors now mismatch:
> 
> image

The colours look fine to me

image

File: apps/web/src/utils/date.ts (hunk 2 above)
Comment (PJColombo) [reply apps/web/src/utils/date.ts:9]: Let's do component and e2e testing for the web app instead

**Pull request merged**